test/python/quantum_info/operators/test_operator.py

```diff
@@ -838,9 +861,108 @@ def test_from_circuit_constructor_empty_layout(self):
         circuit.h(0)
         circuit.cx(0, 1)
         layout = Layout()
-        with self.assertRaises(KeyError):
+        with self.assertRaises(IndexError):
             Operator.from_circuit(circuit, layout=layout)
 
+    def test_from_circuit_single_flat_default_register_transpiled(self):
+        """Test a transpiled circuit with layout set from default register."""
+        circuit = QuantumCircuit(5)
+        circuit.h(3)
+        circuit.cx(3, 4)
+        circuit.cx(3, 2)
+        circuit.cx(3, 0)
+        circuit.cx(3, 1)
+        print(circuit)
+        init_layout = Layout(
+            {
+                circuit.qubits[0]: 3,
+                circuit.qubits[1]: 4,
+                circuit.qubits[2]: 1,
+                circuit.qubits[3]: 2,
+                circuit.qubits[4]: 0,
+            }
+        )
+        tqc = transpile(circuit, initial_layout=init_layout)
+        print(tqc)
+        result = Operator.from_circuit(tqc)
+        print(result)
+        print(Operator(circuit))
```

Comment:
debugging prints here

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/8802/commits/bb24c4770885b5fadd3e13c0db363e0c4fb4ab04